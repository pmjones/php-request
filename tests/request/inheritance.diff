006+ 
007+ Fatal error: Uncaught RuntimeException: MagicSapiRequest::$magicTest is read-only. in /Users/pmjones/Code/pmjones/php-request/src/SapiRequest.php:125
008+ Stack trace:
009+ #0 /Users/pmjones/Code/pmjones/php-request/tests/request/inheritance.php(39): SapiRequest->__set('magicTest', 'baz')
010+ #1 {main}
006- string(3) "baz"
007- string(3) "bat"
008- ok
009- string(16) "RuntimeException"
010- string(38) "CtorSapiRequest::$method is read-only."
011+   thrown in /Users/pmjones/Code/pmjones/php-request/src/SapiRequest.php on line 125