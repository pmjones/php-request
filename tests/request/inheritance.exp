string(3) "PUT"
string(3) "foo"
string(3) "bar"
string(16) "RuntimeException"
string(37) "SubSapiRequest::$method is read-only."
string(3) "baz"
string(3) "bat"
ok
string(16) "RuntimeException"
string(38) "CtorSapiRequest::$method is read-only."