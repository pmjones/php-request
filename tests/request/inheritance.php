<?php
$_SERVER = [
    'HTTP_HOST' => 'example.com',
    'REQUEST_METHOD' => 'PUT',
];
class SubSapiRequest extends SapiRequest {
    public $publicTest;
    protected $protectedTest;
    private $privateTest;
    private $magicGetTest;
    public function protectedTest($a) {
        $this->protectedTest = $a;
        return $this->protectedTest;
    }
    public function parentReadOnlyTest() {
        $this->method = 'PATCH';
    }
}
class MagicSapiRequest extends SapiRequest {
    protected $magicTest;
}
class CtorSapiRequest extends SapiRequest {
    public function __construct(array $globals = null) {
        parent::__construct($globals);
        $this->method = 'FOO';
    }
}
$request = new SubSapiRequest($GLOBALS);
var_dump($request->method);
$request->publicTest = 'foo';
var_dump($request->publicTest);
var_dump($request->protectedTest('bar'));
try {
    $request->parentReadOnlyTest();
} catch( Exception $e ) {
    var_dump(get_class($e), $e->getMessage());
}
$request = new MagicSapiRequest($GLOBALS);
$request->magicTest = 'baz';
var_dump($request->magicTest);
$request->magicTestUndef = 'bat';
var_dump($request->magicTestUndef);

// Make sure cloning keeps readonly
$clone = clone $request;
try {
    $clone->method = 'PUT';
    echo 'fail';
} catch( Exception $e ) {
    echo 'ok';
}
echo PHP_EOL;

// constructor overrides of parent properties not allowed
try {
    $request = new CtorSapiRequest($GLOBALS);
} catch (Exception $e) {
    var_dump(get_class($e), $e->getMessage());
}
