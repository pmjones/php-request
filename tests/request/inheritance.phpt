--TEST--
SapiRequest - inheritance
--FILE--
<?php
$_SERVER = [
    'HTTP_HOST' => 'example.com',
    'REQUEST_METHOD' => 'PUT',
];
class SubSapiRequest extends SapiRequest {
    public $publicTest;
    protected $protectedTest;
    private $privateTest;
    private $magicGetTest;
    public function protectedTest($a) {
        $this->protectedTest = $a;
        return $this->protectedTest;
    }
    public function parentReadOnlyTest() {
        $this->method = 'PATCH';
    }
}
class MagicSapiRequest extends SapiRequest {
    protected $magicTest;
}
class CtorSapiRequest extends SapiRequest {
    public function __construct(array $globals = null) {
        parent::__construct($globals);
        $this->method = 'FOO';
    }
}
$request = new SubSapiRequest($GLOBALS);
var_dump($request->method);
$request->publicTest = 'foo';
var_dump($request->publicTest);
var_dump($request->protectedTest('bar'));
try {
    $request->parentReadOnlyTest();
} catch( Exception $e ) {
    var_dump(get_class($e), $e->getMessage());
}
$request = new MagicSapiRequest($GLOBALS);
$request->magicTest = 'baz';
var_dump($request->magicTest);
$request->magicTestUndef = 'bat';
var_dump($request->magicTestUndef);

// Make sure cloning keeps readonly
$clone = clone $request;
try {
    $clone->method = 'PUT';
    echo 'fail';
} catch( Exception $e ) {
    echo 'ok';
}
echo PHP_EOL;

// constructor overrides of parent properties not allowed
try {
    $request = new CtorSapiRequest($GLOBALS);
} catch (Exception $e) {
    var_dump(get_class($e), $e->getMessage());
}
--EXPECT--
string(3) "PUT"
string(3) "foo"
string(3) "bar"
string(16) "RuntimeException"
string(37) "SubSapiRequest::$method is read-only."
string(3) "baz"
string(3) "bat"
ok
string(16) "RuntimeException"
string(38) "CtorSapiRequest::$method is read-only."
