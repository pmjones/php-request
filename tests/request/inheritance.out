string(3) "PUT"
string(3) "foo"
string(3) "bar"
string(16) "RuntimeException"
string(37) "SubSapiRequest::$method is read-only."

Fatal error: Uncaught RuntimeException: MagicSapiRequest::$magicTest is read-only. in /Users/pmjones/Code/pmjones/php-request/src/SapiRequest.php:125
Stack trace:
#0 /Users/pmjones/Code/pmjones/php-request/tests/request/inheritance.php(39): SapiRequest->__set('magicTest', 'baz')
#1 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/src/SapiRequest.php on line 125